docs/install/Knative-with-any-k8s.md

```diff
@@ -1,89 +1,470 @@
+   ```bash
+   kubectl apply --filename https://raw.githubusercontent.com/knative/serving/{{< version >}}/third_party/contour-latest/contour.yaml
+   ```
 
-## Installing Knative
+2. To configure Knative Serving to use Contour by default:
 
-The following commands install all available Knative components. To customize
-your Knative installation, see
-[Performing a Custom Knative Installation](./Knative-custom-install.md).
+    ```bash
+    kubectl patch configmap/config-network \
+      --namespace knative-serving \
+      --type merge \
+      --patch '{"data":{"ingress.class":"contour.ingress.networking.knative.dev"}}'
+    ```
 
-1. To install Knative, first install the CRDs by running the `kubectl apply`
-   command once with the `-l knative.dev/crd-install=true` flag. This prevents
-   race conditions during the install, which cause intermittent errors:
+3. Fetch the External IP or CNAME
 
    ```bash
-   kubectl apply --selector knative.dev/crd-install=true \
-   --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving.yaml \
-   --filename https://github.com/knative/eventing/releases/download/{{< version >}}/eventing.yaml \
-   --filename https://github.com/knative/serving/releases/download/{{< version >}}/monitoring.yaml
+   kubectl --namespace projectcontour get service envoy-external
    ```
 
-1. To complete the install of Knative and its dependencies, run the
-   `kubectl apply` command again, this time without the `--selector` flag, to
-   complete the install of Knative and its dependencies:
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+
+{{% tab name="Gloo" %}}
+_For a detailed guide on Gloo integration, see [here](https://docs.solo.io/gloo/latest/installation/knative/)_
+
+The following commands install Gloo and enables its Knative integration:
+
+1. Make sure `glooctl` is installed (version 1.3.x and higher recommended):
 
    ```bash
-   kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving.yaml \
-   --filename https://github.com/knative/eventing/releases/download/{{< version >}}/eventing.yaml \
-   --filename https://github.com/knative/serving/releases/download/{{< version >}}/monitoring.yaml
+   glooctl version
    ```
 
+   If it is not installed, follow the instructions [here](https://docs.solo.io/gloo/latest/installation/knative/#install-command-line-tool-cli)
+
+2. Install Gloo and the Knative integration:
+
+    ```bash
+    glooctl install knative --install-knative=false
+    ```
+
+3. Fetch the External IP or CNAME
+
+   ```bash
+   kubectl get svc -ngloo-system knative-external-proxy
+   ```
+
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+
+{{% tab name="Istio" %}}
+The following commands install Istio and enables its Knative integration:
+
+1. [Installing Istio for Knative](./installing-istio.md)
```

Comment:
All of the other guides have commands inline. Linking out to a different document within a tab and expecting the user to come back to the same tab feels like a bad experience. 

Can we have a basic install command and have a `_For a detailed guide on Istio integration, see [here]` like we do for Gloo?

Reply:
This isn't that different from what we have today, so I'd be inclined to leave this as-is and iterate with further incremental improvements.  I'll open an issue and `TODO(#1234)` it.

Reply:
We could also go the route of including the content from the installing istio page here using readfile.

Reply:
The istio page is too large for this section. We would need to slim it down a bit.

Reply:
I am okay with the issue that was created.